CONTRIBUTING.md

```diff
@@ -0,0 +1,147 @@
+
+## How to contribute?
+
+This project uses [Github Flow](https://guides.github.com/introduction/flow/index.html), so all code changes happen through [Pull Requests](https://github.com/jenkins-infra/pipeline-library/pulls). In order to create a Pull Request, you have to:
+
+- Fork the repo and create your branch from master.
+- If you've added code that should be tested, add tests.
+- If you've changed APIs, update the documentation.
+- Ensure the test suite passes.
+- Make sure your code lints.
+- Issue that pull request!
+
+## How will my contribution be evaluated?
+
+As much as all contributions are welcome in the project, please keep in mind that any addition to the source code means some additional code to maintain. While a contribution usually is a one time effort, maintaining a consistent code base is a long running task.
+
+Keeping that in mind, it makes sense to consider that a contribution needs to be evaluated by the team in charge of the repository, not only to consider the quality of what you produces, but also to evaluate if it's reasonable enough for the team to take some commitment on the long term regarding the addition you propose.
+
+This is the reason why we strongly encourage you to start your contribution journey through discussion with the team, to ensure you are not consuming time and producing efforts which wouldn't lead to an actualy integration within the repository. By starting an early discussion, you'll most likely avoid seeing your contribution rejected.
+
+Once your contribution is made though, please keep in mind that the team will need to review your proposal, and might have some questions or even requests for modifications. Please follow the discussions and keep the Pull Request alive for a smooth experience.
+
+## How can I test my contribution?
+
+This project contains some automated tests which are located in the [test folder](https://github.com/jenkins-infra/pipeline-library/tree/master/test/groovy).
+
+They can be executed with [Maven](https://maven.apache.org/) using the following command: `mvn test`.
+
+If you are contributing new features or changes to the code base, please consider implementing some automated tests as well.
+
+## How to setup my environment?
+
+### Signing Commits
+
+Your commits need to be signed and verified by Github. You can achieve this by following [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
+
+*Don't forget to add your GPG key to your Github profile*.
+
+### Technical Requirements
+
+In order to work with that repository you will need:
+
+- OpenJDK 8: https://openjdk.java.net/projects/jdk8/
+- Maven 3.6+: https://maven.apache.org/download.cgi
+- Groovy: https://groovy-lang.org/install.html
+
+An IDE or text editor of your choice, like [Vim](https://www.vim.org/), [Emacs](https://www.vim.org/) or [VS Code](https://www.vim.org/) for example.
```

Comment:
Copy-paste error, or start of flame war? :grin:

Reply:
It was my gentle troll/easter egg for someone to find :p 

Now that you did I'll fix it to the proper links ;)